src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs

```diff
@@ -11,4 +11,351 @@ public class JinjaTemplateAdapter {
             .Replace("{%", "{{")
             .Replace("%}", "}}");
     }
+
+    public static async Async.Task<bool> IsValidScribanNotificationTemplate(IOnefuzzContext context, ILogTracer log, NotificationTemplate template) {
+        try {
+            var (didModify, _) = template switch {
+                TeamsTemplate => (false, template),
+                AdoTemplate adoTemplate => await ConvertToScriban(adoTemplate, attemptRender: true, context, log),
+                GithubIssuesTemplate githubTemplate => await ConvertToScriban(githubTemplate, attemptRender: true, context, log),
+                _ => throw new ArgumentOutOfRangeException(nameof(template), "Unexpected notification template type")
+            };
+
+            if (!didModify) {
+                return true;
+            }
+            return false;
+        } catch (Exception e) {
+            log.Exception(e);
+            return false;
+        }
+    }
+
+    public static async Async.Task<TemplateValidationResponse> ValidateScribanTemplate(IOnefuzzContext context, ILogTracer log, TemplateRenderContext? renderContext, string template) {
+        var instanceUrl = context.ServiceConfiguration.OneFuzzInstance!;
+
+        var (renderer, templateRenderContext) = await GenerateTemplateRenderContext(context, log, renderContext);
+
+        var renderedTemaplate = await renderer.Render(template, new Uri(instanceUrl), strictRendering: true);
+
+        return new TemplateValidationResponse(
+            renderedTemaplate,
+            templateRenderContext
+        );
+    }
+
+    private static async Async.Task<(NotificationsBase.Renderer, TemplateRenderContext)> GenerateTemplateRenderContext(IOnefuzzContext context, ILogTracer log, TemplateRenderContext? templateRenderContext) {
```

Comment:
This function was copied directly from `src/ApiService/ApiService/Functions/ValidateScriban.cs`